### PR TITLE
add examples in tiflash docs

### PR DESCRIPTION
**Pull request by joey-yez**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/tiflash-supported-pushdown-calculations.md [hunk 1]
```diff
@@ -51,3 +51,57 @@ TiFlash 支持部分算子的下推，支持的算子如下：
     * SECOND
 
 如查询遇到不支持的下推计算，则需要依赖 TiDB 完成剩余计算，可能会很大程度影响 TiFlash 加速效果。对于暂不支持的算子/表达式，将会在后续版本中陆续支持。
+
+## 示例
```
Comment (zanmato1984) [inline tiflash/tiflash-supported-pushdown-calculations.md:55]: 建议补充一个如何排查没有下推算子、表达式的例子：1. 先检查执行计划发现不是 tiflash；2. show warnings 查看具体没有下推的表达式和算子。

Comment (joey-yez) [reply tiflash/tiflash-supported-pushdown-calculations.md:55]: 已经增加示例3

File: tiflash/tiflash-results-materialization.md [hunk 2]
```diff
@@ -66,4 +66,37 @@ SELECT app_name, country FROM t1;
...
-    * 请基于测试和具体情况做出合理选择。
\ No newline at end of file
+    * 请基于测试和具体情况做出合理选择。
+
+## 示例
+
+数据定义：
+```sql
+CREATE TABLE detail_data (
+    ts DATETIME,                -- 详单的时间
```
Comment (shichun-0415) [inline tiflash/tiflash-results-materialization.md:76]: "详单“是指”查询的订单“吗？

File: tiflash/tiflash-results-materialization.md [hunk 3]
```diff
@@ -66,4 +66,37 @@ SELECT app_name, country FROM t1;
...
+ALTER TABLE detail_data SET TIFLASH REPLICA 1;
+ALTER TABLE daily_data SET TIFLASH REPLICA 1;
+
+-- ... (detail_data 表不断增加数据)
+```
+
+每日分析数据保存：
+```sql
+INSERT INTO daily_data (rec_date, customer_id, daily_fee)
+SELECT DATE(ts), customer_id, sum(detail_fee) FROM detail_data WHERE DATE(ts) = CURRENT_DATE() GROUP BY DATE(ts), customer_id;
```
Comment (shichun-0415) [inline tiflash/tiflash-results-materialization.md:94]:
```suggestion
每日分析数据保存：

```sql
INSERT INTO daily_data (rec_date, customer_id, daily_fee);
SELECT DATE(ts), customer_id, sum(detail_fee) FROM detail_data;
WHERE DATE(ts) = CURRENT_DATE() GROUP BY DATE(ts), customer_id;
```

File: tiflash/tiflash-supported-pushdown-calculations.md [hunk 4]
```diff
@@ -51,3 +51,99 @@ TiFlash 支持部分算子的下推，支持的算子如下：
...
+在该查询中，将表达式 id + a 下推到 TiFlash 对数据进行计算，可以提前进行计算，减少网络传输数据量，进而减少网络传输开销，加速整体计算性能。
+
+### 示例 3：下推限制
+
+```sql
+CREATE TABLE t(id INT PRIMARY KEY, a INT);
+ALTER TABLE t SET TIFLASH REPLICA 1;
+INSERT INTO t(id,a) VALUES (1,2),(2,4),(11,2),(12,4),(13,4),(14,7);
+
+EXPLAIN SELECT id FROM t where TIME(now()+ a) < '12:00:00';
```
Comment (shichun-0415) [inline tiflash/tiflash-supported-pushdown-calculations.md:116]:
```suggestion
EXPLAIN SELECT id FROM t WHERE TIME(now()+ a) < '12:00:00';
```

File: tiflash/tiflash-supported-pushdown-calculations.md [hunk 5]
```diff
@@ -51,3 +51,99 @@ TiFlash 支持部分算子的下推，支持的算子如下：
...
+| Projection_4                | 4.80    | root         |               | test.t.id                                                                                        |
+| └─Selection_6               | 4.80    | root         |               | lt(cast(time(cast(plus(20230110083056, test.t.a), var_string(20))), var_string(10)), "12:00:00") |
+|   └─TableReader_11          | 6.00    | root         |               | data:ExchangeSender_10                                                                           |
+|     └─ExchangeSender_10     | 6.00    | mpp[tiflash] |               | ExchangeType: PassThrough                                                                        |
+|       └─TableFullScan_9     | 6.00    | mpp[tiflash] | table:t       | keep order:false, stats:pseudo                                                                   |
++-----------------------------+---------+--------------+---------------+--------------------------------------------------------------------------------------------------+
+5 rows in set, 3 warnings (0.20 sec)
+
```
Comment (shichun-0415) [inline tiflash/tiflash-supported-pushdown-calculations.md:128]:
```suggestion
```

File: tiflash/tiflash-supported-pushdown-calculations.md [hunk 6]
```diff
@@ -51,3 +51,99 @@ TiFlash 支持部分算子的下推，支持的算子如下：
...
+```
+
+通过分析执行计划，可以发现该查询在执行时，只在 TiFlash 中进行了 TableFullScan，其他的函数计算、过滤均在 root 进行，并未下推至 TiFlash。
```
Comment (shichun-0415) [inline tiflash/tiflash-supported-pushdown-calculations.md:131]:
```suggestion
分析执行计划可以发现，该查询在执行时只在 TiFlash 中进行了 TableFullScan，其他的函数计算和过滤均在 `root` 进行，并未下推至 TiFlash。
```

File: tiflash/tiflash-supported-pushdown-calculations.md [hunk 7]
```diff
@@ -51,3 +51,99 @@ TiFlash 支持部分算子的下推，支持的算子如下：
...
+
+通过以下命令查找不能下推的算子、表达式。
```
Comment (shichun-0415) [inline tiflash/tiflash-supported-pushdown-calculations.md:133]:
```suggestion
执行以下命令，可以查找不能下推的算子和表达式。
```

File: tiflash/tiflash-supported-pushdown-calculations.md [hunk 8]
```diff
@@ -51,3 +51,99 @@ TiFlash 支持部分算子的下推，支持的算子如下：
...
+
++---------+------+------------------------------------------------------------------------------------------------------------------------------------+
+| Level   | Code | Message                                                                                                                            |
++---------+------+------------------------------------------------------------------------------------------------------------------------------------+
+| Warning | 1105 | Scalar function 'time'(signature: Time, return type: time) is not supported to push down to storage layer now.                     |
+| Warning | 1105 | Scalar function 'cast'(signature: CastDurationAsString, return type: var_string(10)) is not supported to push down to tiflash now. |
+| Warning | 1105 | Scalar function 'cast'(signature: CastDurationAsString, return type: var_string(10)) is not supported to push down to tiflash now. |
++---------+------+------------------------------------------------------------------------------------------------------------------------------------+
+3 rows in set (0.18 sec)
+
```
Comment (shichun-0415) [inline tiflash/tiflash-supported-pushdown-calculations.md:146]:
```suggestion
```

File: tiflash/tiflash-supported-pushdown-calculations.md [hunk 9]
```diff
@@ -51,3 +51,99 @@ TiFlash 支持部分算子的下推，支持的算子如下：
...
+```
+
+可以看出，该查询的表达式无法完全下推至 TiFlash，是因为 Time 函数和 Cast 函数无法下推至 TiFlash。
```
Comment (shichun-0415) [inline tiflash/tiflash-supported-pushdown-calculations.md:149]:
```suggestion
可以看出，该查询的表达式无法完全下推至 TiFlash，因为 `Time` 函数和 `Cast` 函数无法下推至 TiFlash。
```

File: tiflash/tiflash-results-materialization.md [hunk 10]
```diff
@@ -66,4 +66,37 @@ SELECT app_name, country FROM t1;
...
     * 当“写事务”较大时，例如接近 1 GiB，建议控制并发不超过 10。
     * 当“写事务”较小时，例如小于 100 MiB，建议控制并发不超过 30。
-    * 请基于测试和具体情况做出合理选择。
\ No newline at end of file
+    * 请基于测试和具体情况做出合理选择。
+
+## 示例
+
+数据定义：
+```sql
```
Comment (shichun-0415) [inline tiflash/tiflash-results-materialization.md:74]:
```suggestion
数据定义：

```sql
```
”数据定义“有点简略，可否完整介绍下这个示例？

File: tiflash/tiflash-results-materialization.md (hunk 2 above)
Comment (joey-yez) [reply tiflash/tiflash-results-materialization.md:77]: 就是指每一笔费用，具体精确到对应的 秒，相对后面汇总的单日的 fee 的记录。

**Review by zanmato1984 — APPROVED** [review]
LGTM

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Pull request merged**